tests/Unit/Evolution/Systems/NewtonianEuler/BoundaryConditions/DirichletAnalytic.py

```diff
@@ -0,0 +1,263 @@
+                              _soln_wave_vector(dim), _soln_pressure,
+                              _soln_adiabatic_index, _soln_perturbation_size)
+    return flux.energy_density_flux_impl(
+        soln_momentum_density(face_mesh_velocity,
+                              outward_directed_normal_covector, coords, time,
+                              dim),
+        soln_energy_density(face_mesh_velocity,
+                            outward_directed_normal_covector, coords, time,
+                            dim), velocity, pressure)
+
+
+def soln_velocity(face_mesh_velocity, outward_directed_normal_covector, coords,
+                  time, dim):
+    return hydro.spatial_velocity(coords, time, _soln_mean_velocity(dim),
+                                  _soln_wave_vector(dim), _soln_pressure,
+                                  _soln_adiabatic_index,
+                                  _soln_perturbation_size)
+
+
+def soln_specific_internal_energy(face_mesh_velocity,
+                                  outward_directed_normal_covector, coords,
+                                  time, dim):
+    return hydro.specific_internal_energy(coords, time,
+                                          _soln_mean_velocity(dim),
+                                          _soln_wave_vector(dim),
+                                          _soln_pressure,
+                                          _soln_adiabatic_index,
+                                          _soln_perturbation_size)
+
+
+def data_error(face_mesh_velocity, outward_directed_normal_covector, coords,
+               time, dim):
+    return None
+
+
+_data_adiabatic_index = 1.4
+_data_strip_bimedian_height = 0.5
+_data_strip_thickness = 0.5
+_data_strip_density = 2.0
+_data_strip_velocity = 0.5
+_data_background_density = 1.0
+_data_background_velocity = -0.5
+_data_pressure = 2.5
+_data_perturb_amplitude = 0.1
+_data_perturb_width = 0.035355339059327376
+
+
+def data_mass_density(face_mesh_velocity, outward_directed_normal_covector,
+                      coords, time, dim):
+    if np.abs(coords[-1] -
+              _data_strip_bimedian_height) < 0.5 * _data_strip_thickness:
+        return _data_strip_density
+    else:
+        return _data_background_density
+
+
+def data_velocity(face_mesh_velocity, outward_directed_normal_covector, coords,
+                  time, dim):
+    velocity = np.zeros([dim])
+    if np.abs(coords[-1] -
+              _data_strip_bimedian_height) < 0.5 * _data_strip_thickness:
+        velocity[0] = _data_strip_velocity
+    else:
+        velocity[0] = _data_background_velocity
+
+    one_over_two_sigma_squared = 0.5 / (_data_perturb_width)**2
+    strip_lower_bound = (_data_strip_bimedian_height -
+                         0.5 * _data_strip_thickness)
+    strip_upper_bound = (_data_strip_bimedian_height +
+                         0.5 * _data_strip_thickness)
+    velocity[dim -
+             1] = np.exp(-one_over_two_sigma_squared *
```

Comment:
It is possible that yapf would format this less weird if you wrapped the whole right-hand side in parens